superset-frontend/src/chart/chartAction.js

```diff
@@ -394,15 +394,17 @@ export function exploreJSON(
     const chartDataRequestCaught = chartDataRequest
       .then(({ response, json }) => {
         if (isFeatureEnabled(FeatureFlag.GLOBAL_ASYNC_QUERIES)) {
-          // deal with getChartDataRequest transforming the response data
-          const result = 'result' in json ? json.result[0] : json;
+          const result = 'result' in json ? json.result : json;
           switch (response.status) {
             case 200:
-              // Query results returned synchronously, meaning query was already cached.
-              return Promise.resolve([result]);
+              return Promise.resolve(result);
             case 202:
-              // Query is running asynchronously and we must await the results
-              return waitForAsyncData(result);
+              if (shouldUseLegacyApi(formData)) {
+                // Query is running asynchronously and we must await the results
+                return waitForAsyncData(result[0]);
+              } else {
+                return waitForAsyncData(result);
+              }
```

Comment:
```suggestion
              if (shouldUseLegacyApi(formData)) {
                // Query is running asynchronously and we must await the results
                return waitForAsyncData(result[0]);
              }
              return waitForAsyncData(result);
```

Reply:
Sure! I will upload the video as soon as I can :):)

Reply:
@geido Thanks for your review:) I uploaded the videos and information for testing:)